fedimint-server/src/net/peers.rs

```diff
@@ -85,36 +82,37 @@ struct PeerConnectionStateMachine<M> {
 /// The default values are in the order of seconds
 #[derive(Debug, Clone, Copy)]
 pub struct DelayCalculator {
-    scaling_factor: f64,
+    max_fail_reconnect_counter: u64,
 }
 
 impl DelayCalculator {
-    fn new() -> Self {
-        Self {
-            scaling_factor: 1.0,
-        }
-    }
+    /// Maximum connection failures we consider for our back-off strategy,
+    /// production default
+    const PROD_MAX_FAIL_RECONNECT_COUNTER: u64 = 300;
```

Comment:
Should this be a `Duration` bound instead, so it's more clear?

If someone changes the `scaling_factor` then `300` might end up being too high.

Reply:
```
> for i in $(seq 1 20) ; echo "$i: $(math "(4 ^ $i) / 1000" )ms" ; end
1: 0.004ms
2: 0.016ms
3: 0.064ms
4: 0.256ms
5: 1.024ms
6: 4.096ms
7: 16.384ms
8: 65.536ms
9: 262.144ms
10: 1048.576ms
11: 4194.304ms
12: 16777.216ms
13: 67108.864ms
14: 268435.456ms
15: 1073741.824ms
16: 4294967.296ms
17: 17179869.184ms
18: 68719476.736ms
19: 274877906.944ms
20: 1099511627.776ms
```

I think this is something that we could just have both in tests and in prod at the same time (with a cap of 10s maybe).

5 very quick ones, another 5 quite fast as well, but then quickly straight to the ceiling.

Reply:
I've created https://github.com/fedimint/fedimint/pull/2285 using these ideas